engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -750,4 +760,73 @@ public Map<String, Object> getTableAttributes(
         }
         return attributes;
     }
+
+    @Override
+    public long estimatePushdownFilterCost(
+            final WhereFilter filter,
+            final RowSet selection,
+            final RowSet fullSet,
+            final boolean usePrev,
+            final FilterContext context) {
+
+        if (includedLocations().isEmpty()) {
+            return 0; // No locations, no cost
+        }
+
+        // Test a few locations and assume the rest are similar (or no worse)
+        final long costPerLocation = includedLocations().stream().limit(PUSH_DOWN_LOCATIONS_TO_TEST).mapToLong(
+                location -> location.estimatePushdownFilterCost(filter, selection, fullSet, usePrev, context))
+                .reduce(Long::max).orElse(Long.MAX_VALUE);
+
+        return LongMath.saturatedMultiply(includedLocations().size(), costPerLocation);
```

Comment:
Do we want to get fancy with multiplication here?  Like if you have a parquet table with stats, do you really want to have metadata cost * locations size or just keep it in the same metadata cost bucket?